subsys/bluetooth/audio/stream.c

```diff
@@ -1071,6 +1034,54 @@ int bt_audio_unicast_group_delete(struct bt_audio_unicast_group *unicast_group)
 
 #endif /* CONFIG_BT_AUDIO_UNICAST_CLIENT */
 
+int bt_audio_stream_start(struct bt_audio_stream *stream)
+{
+	uint8_t state;
+	uint8_t role;
+	int err;
+
+	BT_DBG("stream %p", stream);
+
+	CHECKIF(stream == NULL || stream->ep == NULL || stream->conn == NULL) {
+		BT_DBG("Invalid stream");
+		return -EINVAL;
+	}
+
+	role = stream->conn->role;
+	if (role != BT_HCI_ROLE_CENTRAL) {
+		BT_DBG("Invalid conn role: %u, shall be central", role);
+		return -EINVAL;
+	}
+
+	state = stream->ep->status.state;
+	switch (state) {
+	/* Valid only if ASE_State field = 0x03 (Enabling) */
+	case BT_AUDIO_EP_STATE_ENABLING:
+		break;
+	default:
+		BT_ERR("Invalid state: %s", bt_audio_ep_state_str(state));
+		return -EBADMSG;
+	}
+
+	role = stream->conn->role;
+	if (IS_ENABLED(CONFIG_BT_AUDIO_UNICAST_CLIENT) &&
+	    role == BT_HCI_ROLE_CENTRAL) {
+		err = bt_unicast_client_start(stream);
+	} else if (IS_ENABLED(CONFIG_BT_AUDIO_UNICAST_SERVER) &&
+		   role == BT_HCI_ROLE_PERIPHERAL) {
```

Comment:
How can this ever happen?  If role is PERIPHERAL, the function will return in line 1053.
Should lines 1050 - 1054 be deleted?

Reply:
Good catch! Yes that role check should be removed